### DISCUSSION TITLE
v0.1.6

### DISCUSSION THREAD

**Discussion by Luosuu** [ByteDance-Seed/VeOmni#432] [Announcements]
## Highlights

* Now we gradually migrate VeOmni modeling to patching the HuggingFace modeling so that users can easily know what is different from original HuggingFace implementation. Currently we migrate dense models first. @Coach257 @FoolPlayer 
  * In the following, we will work on generating modeling like HuggingFace, which will happen at the same time with HuggingFace Transformers v5 upgrade. @piyifan123 
* Support Qwen3-Omni-MoE by @Crystal-jiang .
* Now VeOmni does not override HuggingFace Transformers `ALL_ATTENTION_FUNCTION` registry. To ensure backward compatibility, when `flash_attention_2/3` is passed to model arguments, they will be replaced by veomni flash attention key names. @Luosuu 
* Support padding for packed input when `rmpad_with_pos_ids`, which will eliminate expensive Triton compiling for varying input size and pave the way for torch.compile integration.
* Support `torchcodec` -based video preprocessing by @TimYangst 
* Many fixes.

## What's Changed
* [docs] fix: Optimize document links in Markdown rendering by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/380
* [config] feat:  add MFU calculation for qwen3_vl_moe by @ZhuYajun-AI in https://github.com/ByteDance-Seed/VeOmni/pull/385
* [data, model] feat: support Qwen3-VL textual token-based time encoding by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/386
* [data,ci] test: enhance video_utils test suite with robust validation and benchmarks by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/375
* [data,ci,docs] feat: add torchcodec-based video processing with ffmpeg support and comprehensive testing by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/221
* [perf, dist] feat: add zero2 in fsdp1 and use_orig_params configurable by @zhtao303 in https://github.com/ByteDance-Seed/VeOmni/pull/382
* [model] fix: Fused operator fix for qwen3vl by @phdddd in https://github.com/ByteDance-Seed/VeOmni/pull/378
* [docs] feat: add async doc in ulysses.md by @zbian99 in https://github.com/ByteDance-Seed/VeOmni/pull/388
* [data] feat: add data collators for embedding classification. by @yiwzhao in https://github.com/ByteDance-Seed/VeOmni/pull/376
* [model] chore: add moe split script by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/390
* [task] feat: train qwen2.5 omni by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/396
* [model] refactor: change to model patch in qwen3 by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/392
* [docker] fix: update npu dockerfile with ffmpeg by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/398
* [config] feat: refactor args to support multi-level config by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/397
* [docs] fix: update async ulysses document by @zbian99 in https://github.com/ByteDance-Seed/VeOmni/pull/394
* [model] fix: add fa3 for Qwen3VL vision attention SP path by @Luosuu in https://github.com/ByteDance-Seed/VeOmni/pull/400
* [model] feat: patch qwen3vlmoe; qwen3vl; qwen25vl by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/399
* [model] fix: qwen25vl_config by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/404
* [model] fix: fsdp1 load model weights by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/403
* [config] fix: remove `None` option from choice type args by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/405
* [misc] feat: add live star history in readme by @Luosuu in https://github.com/ByteDance-Seed/VeOmni/pull/406
* [config, model] chore: avoid polluting huggingface transformer attention registry while keeping job config backward compatiable by @Luosuu in https://github.com/ByteDance-Seed/VeOmni/pull/407
* [docs] fix: correct typos in documentation by @zwhe99 in https://github.com/ByteDance-Seed/VeOmni/pull/411
* [data] feat: decouple rmpad and dyn_bsz by @yangtian6781 in https://github.com/ByteDance-Seed/VeOmni/pull/408
* [model] feat: refactor the rot_pos_emb and fast_pos_embed_interpolate funcs in modeling_qwen_vl by @lipengfei1409 in https://github.com/ByteDance-Seed/VeOmni/pull/391
* [ci] fix: use wheel URLs to avoid build dependencies in CI  by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/417
* [model] fix: remove the check that self.training==True for SP by @A1waysBeenHere in https://github.com/ByteDance-Seed/VeOmni/pull/421
* [model] feat: Add pure HuggingFace version of Qwen3-Omni-MoE by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/422
* [data, perf, ops] feat: option to pad packed input to a fixed shape for text-only models by @Luosuu in https://github.com/ByteDance-Seed/VeOmni/pull/410
* [ops, perf] feat: Liger-Kernel is now available for NPU by @zheliuyu in https://github.com/ByteDance-Seed/VeOmni/pull/415
* [model] feat: Support qwen3-omni-moe model by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/409
* [dist] feat:  No resharding enabled for accelerated small model training by @yangtian6781 in https://github.com/ByteDance-Seed/VeOmni/pull/413
* [model] fix: qwen_vl rope index by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/430
* [model] refactor: change dense llm to patch style by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/431
* [perf] fix: The H200 compute power is being recognized as H20 by @HSYZhang in https://github.com/ByteDance-Seed/VeOmni/pull/428

## New Contributors
* @ZhuYajun-AI made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/385
* @zhtao303 made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/382
* @zwhe99 made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/411
* @yangtian6781 made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/408
* @lipengfei1409 made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/391
* @A1waysBeenHere made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/421
* @zheliuyu made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/415
* @HSYZhang made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/428

**Full Changelog**: https://github.com/ByteDance-Seed/VeOmni/compare/v0.1.5...v0.1.6

<hr /><em>This discussion was created from the release <a href='https://github.com/ByteDance-Seed/VeOmni/releases/tag/v0.1.6'>v0.1.6</a>.</em>
